apps/src/code-studio/hiddenStageRedux.js

```diff
@@ -34,42 +34,74 @@ const HiddenState = Immutable.Record({
   })
 });
 
+/**
+ * Validates that we never have multiple stagesBySection if we have STUDENT_SECTION_ID
+ * @throws If new state is invalid
+ */
+function validateSectionIds(state) {
+  if (state.getIn(['stagesBySection', STUDENT_SECTION_ID]) &&
+      state.get('stagesBySection').size > 1) {
+    throw new Error('Should never have STUDENT_SECTION_ID alongside other sectionIds');
+  }
+}
+
 /**
  * hidden stage reducer
  * Mapping of stage ids to bools indicating whether it's hidden or not
  */
 export default function reducer(state = new HiddenState(), action) {
+  if (action.type === SET_HIDDEN_STAGES) {
+    const { stagesPerSection, hideableStagesAllowed } = action;
```

Comment:
Please change `stagesPerSection` to `hiddenStagesPerSection`, to differentiate it from `stagesBySection` a little, and add a comment explaining that it is a map from section to a list of hidden stage ids.

Reply:
the docs for `stagesBySection` are a little misleading too: `mapping of section id to hidden stages for that section`. would you mind changing that to something like `mapping from section id to a map indicating whether each of its stages is hidden or not` ?